modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -190,9 +193,41 @@ public synchronized void undeploy(ObjectDefinition objectDefinition) {
 				return;
 			}
 
+			boolean unregisterSystemObjectDefinition = true;
+
 			JaxRsApplicationDescriptor jaxRsApplicationDescriptor =
 				systemObjectDefinitionMetadata.getJaxRsApplicationDescriptor();
 
+			Map<Long, List<ServiceRegistration<?>>> serviceRegistrationsMap =
+				_getServiceRegistrationsMap(
+					jaxRsApplicationDescriptor.getRESTContextPath());
+
+			if (serviceRegistrationsMap != null) {
+				List<ServiceRegistration<?>> serviceRegistrations =
+					serviceRegistrationsMap.remove(
+						objectDefinition.getCompanyId());
+
+				if (serviceRegistrations != null) {
```

Comment:
I think this code is duplicated. Could we refactor this to avoid duplicating it?